sound/soc/sof/xtensa/core.c

```diff
@@ -81,41 +81,47 @@ static const struct xtensa_exception_cause xtensa_exception_causes[] = {
 };
 
 /* only need xtensa atm */
-static void xtensa_dsp_oops(struct snd_sof_dev *sdev, void *oops)
+static void xtensa_dsp_oops(struct snd_sof_dev *sdev, const char *level, void *oops)
```

Comment:
im not sure how this will be perceived upstream either. If we have a DSP panic, it should always be printed with the error level or at least a warning if it is during boot with an attempted recovery. I dont appreciate the idea of hiding it and only displaying it for the developers when a debug flag is set.

Reply:
@ranj063 we had an earlier discussion about this in #3238 (I thought it was already linked but it was not).

This is partly similar to `printk_ratelimit` but controllable at run-time with a module parameter and with a different log level for the repetition.

Reply:
@ranj063, with the latest version of this PR we are _always_ print the DSP dump in KERN_ERR level from the DSP panic path and (while I don't like it that much) it is going to be printed every time we receive a panic, but the message is going to be different to indicated that there will be recovery attempt made.

The printk level is needed for the other path which will print dumps in KERN_DEBUG in case the print was optional, but we want to see all of them.

It took quite a while to align the conflicting requirements and have a compromise which is acceptable.